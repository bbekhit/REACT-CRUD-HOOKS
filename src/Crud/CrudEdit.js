import React from "react";
import CrudForm from "./CrudForm";
import { connect } from "react-redux";
import { editNote } from "../actions/crudActions";

const CrudEdit = ({ note, editNote, history }) => {
  const onSubmit = data => {
    let notesData = JSON.parse(localStorage.getItem("notes"));
    notesData = notesData.map(item =>
      item.id === note.id ? (item = data) : item
    );
    localStorage.setItem("notes", JSON.stringify(notesData));
    editNote(note.id, data);
    history.push("/");
  };
  return (
    <div>
      <CrudForm note={note} onSubmit={data => onSubmit(data)} />
    </div>
  );
};

const mapStateToProps = (state, props) => ({
  note: state.notes.find(item => item.id === props.match.params.id)
});

export default connect(mapStateToProps, { editNote })(CrudEdit);
