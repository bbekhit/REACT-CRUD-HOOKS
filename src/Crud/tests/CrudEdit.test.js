import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { CrudEdit } from "../CrudEdit";
import notes from "./mockData";

describe("Crud Add", () => {
  let wrapper;
  let editNote;
  let history;
  beforeEach(() => {
    editNote = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
      <CrudEdit
        note={notes[0]}
        editNote={editNote}
        history={history}
        notesData={notes}
      />
    );
  });
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("handles onSubmit correctly", () => {
    wrapper.find("#edit").prop("onSubmit")(notes[0]);
    expect(history.push).toHaveBeenLastCalledWith("/");
    expect(editNote).toHaveBeenLastCalledWith(notes[0].id, notes[0]);
  });
});
