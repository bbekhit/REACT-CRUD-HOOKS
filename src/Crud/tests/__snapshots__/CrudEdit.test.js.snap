// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crud Add renders correctly 1`] = `
<div>
  <Connect(withRouter(CrudForm))
    id="edit"
    note={
      Object {
        "body": "Body-1",
        "id": 1,
        "title": "Title-1",
      }
    }
    onSubmit={[Function]}
  />
</div>
`;
